src/main/native/SymmetricCipher.c

```diff
@@ -40,7 +40,12 @@ typedef struct OCKCipher
   int copy_context;
 } OCKCipher;
 
-PFI KMC; // z_kmc_native function pointer; only available on some hardware; might be null
+typedef size_t UDATA;
+typedef signed char* arr;
+
+// Pointers of functions that are only available on some hardware (might be null)
+typedef void (*KMC_FuncPtr)(unsigned char*, unsigned char*, UDATA*, long, UDATA*);
```

Comment:
Space before `*`.

Reply:
A new header file named zHardwareFunctions.h was created. In this file there should be a space before each * now including this definition.